src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1292,43 +1292,55 @@ export default class AircraftModel {
     /**
      * Initialize all autopilot systems after being given an IFR clearance to destination and execute takeoff.
      *
-     * @for Pilot
+     * @for AircraftModel
      * @method takeoff
      * @param runway {RunwayModel} the runway taking off on
-     * @param initialAltitude {number} the altitude aircraft can automatically climb to at this airport
      */
-    takeoff(runway, initialAltitude) {
+    takeoff(runway) {
         const cruiseSpeed = this.model.speed.cruise;
+        const initialAltitude = this.fms.getInitialClimbClearance();
 
+        this._prepareMCPforTakeoff(initialAltitude, runway.angle, cruiseSpeed);
+        this.setFlightPhase(FLIGHT_PHASE.TAKEOFF);
+        EventBus.trigger(AIRCRAFT_EVENT.TAKEOFF, this, runway);
+
+        this.takeoffTime = TimeKeeper.accumulatedDeltaTime;
+        runway.lastDepartedAircraftCallsign = this.callsign;
+    }
+
+    /**
+     * Initialize all autopilot systems for takeoff.
+     *
+     * @for AircraftModel
+     * @method _prepareMCPforTakeoff
+     * @param altitude {number}
+     * @param heading {number}
+     * @param speed {number}
+     */
+    _prepareMCPforTakeoff(altitude, heading, speed) {
```

Comment:
I favor this casing personally, because MCP is an acronym, but we've stuck with forcible camel-casing stuff like this (why you'll also see `Fms` throughout the code). So this should be changed to `_prepareMcpForTakeoff()`

Reply:
done